### ISSUE TITLE
Expressions defined in Mapping annotations generate invalid code in v4

### ISSUE THREAD

**Issue by msfjarvis** [mcarleio/konvert#99]
With the following mappings defined

```kotlin
mappings = [
  Mapping(target = "submitter", expression = "it.submitterName"),
  Mapping(target = "commentCount", expression = "it.commentCount ?: 0"),
]
```

The generated code has changed in v4 in the below manner

```diff
@@ -23,8 +23,8 @@
   title = savedPost.title,
   url = savedPost.url,
   description = savedPost.description,
-  commentCount = savedPost.let { it.commentCount ?: 0 },
+  commentCount = savedPost.let { let { it.commentCount ?: 0 } },
   commentsUrl = savedPost.commentsUrl,
-  submitter = savedPost.let { it.submitterName },
+  submitter = savedPost.let { let { it.submitterName } },
   tags = savedPost.tags
 )
```

Which results in the `it` implicit field from `savedPost.let` being shadowed by the inner `let` that makes it bind to the class being konverted to rather than the one it's being konverted from.

I was able to fix the first mapping by using `source` instead of `expression` but there's no way to make the second one work.

You can find the source of the file [here](https://github.com/msfjarvis/compose-lobsters/blob/76dd90a28747d9ad1b4e0107de8980e104f02e34/model/src/main/kotlin/dev/msfjarvis/claw/model/UIPost.kt), the `SavedPost` class is generated by a different processor and looks like this:

```kotlin
public data class SavedPost(
  public val shortId: String,
  public val title: String,
  public val url: String,
  public val createdAt: String,
  public val commentCount: Int?,
  public val commentsUrl: String,
  public val submitterName: String,
  public val tags: List<String>,
  public val description: String,
)
```

**Comment by mcarleio** [comment]
That duplicate `let` looks very unintended, I will have a look later and write a unit test for that.

**Comment by mcarleio** [comment]
fixed with 4.0.1

**Comment by msfjarvis** [comment]
[Can confirm](https://msfjarvis.dev/g/compose-lobsters/45a26854cd52), thanks for the quick fix!
